### PR TITLE
Gossiper: Isolate SynchronizeNode from query loop

### PR DESCRIPTION
**Pull request by cfromknecht**
This commit refactors the SyncronizeNode logic such that it can be called without interacting with the gossiper's main execution loop. This method does not require access to any of the gossiper's internal state, making the change fairly straightforward. The primary motivation behind this change is to minimize the likelihood of deadlock scenarios between the gossiper and server.

### CODE REVIEW

**Comment by jimpo** [comment]
LGTM

File: peer.go [hunk 1]
```diff
@@ -1099,7 +1099,9 @@ func (p *peer) PingTime() int64 {
 func (p *peer) queueMsg(msg lnwire.Message, doneChan chan struct{}) {
 	select {
 	case p.outgoingQueue <- outgoinMsg{msg, doneChan}:
+		peerLog.Infof("queued msg successfully")
```
Comment (halseth) [inline peer.go]: `Debugf`?

Comment (halseth) [reply peer.go]: I guess this will happen quite often, so maybe even `Trace` 😛

Comment (Roasbeef) [reply peer.go]: Guessing this was left over as part of a debugging session? It should either be removed, or be demoted to a `Tracef` logging level.

Comment (cfromknecht) [reply peer.go]: Ignore the second commit, was using the latter commit to try and reproduce bugs on travis. Reverting to original PR

**Pull request merged**